modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/slot/SlotManagerClusterMode.java

```diff
@@ -123,30 +88,33 @@
      */
     private Set<String> queuesToRecover;
 
-    /**
-     * Creates a slot manager for cluster mode slot coordination
-     */
-    private SlotManagerClusterMode() {
+    private SlotAgent slotAgent;
 
-        HazelcastAgent hazelcastAgent = HazelcastAgent.getInstance();
-        removedNode = "";
-        //Initialize distributed maps used in this class
-        unAssignedSlotMap = hazelcastAgent.getUnAssignedSlotMap();
-        overLappedSlotMap = hazelcastAgent.getOverLappedSlotMap();
-        slotIDMap = hazelcastAgent.getSlotIdMap();
-        queueToLastAssignedIDMap = hazelcastAgent.getLastAssignedIDMap();
-        nodeToLastPublishedIDMap = hazelcastAgent.getLastPublishedIDMap();
-        slotAssignmentMap = hazelcastAgent.getSlotAssignmentMap();
+    private AndesContextStore andesContextStore;
 
+    private SlotManagerClusterMode() {
+
+        andesContextStore = AndesContext.getInstance().getAndesContextStore();
         nodeInformedSlotDeletionSafeZones = new HashMap<>();
-        slotRecoveryScheduled = new AtomicBoolean(false);
 
         //start a thread to calculate slot delete safe zone
-        //TODO: use a common  thread pool for tasks like this?
         slotDeleteSafeZoneCalc = new SlotDeleteSafeZoneCalc(SAFE_ZONE_EVALUATION_INTERVAL);
         new Thread(slotDeleteSafeZoneCalc).start();
 
+        if(AndesConfigurationManager.readValue(AndesConfiguration.SLOT_MANAGEMENT_STORAGE) != null &&
+            AndesConfigurationManager.readValue(AndesConfiguration.SLOT_MANAGEMENT_STORAGE).equals("RDBMS")) {
+            //Use RDBMS slot information storing
+            slotAgent = new RDBMSAgent();
+
+        } else {
+            //Use Hazelcast slot information storing
+            slotAgent = HazelcastAgent.getInstance();
+
+        }
+
+        removedNode = "";
```

Comment:
Logic used here was changed to Initialize the current node ID as the removed node. This is to avoid slot recovery task adding an empty node ID to the map used to calculate safe zone.